src/shared/components/results/ResultsView.tsx

```diff
@@ -1,46 +1,88 @@
   UniProtkbAPIModel,
-} from '../../adapters/uniProtkbConverter';
-import { UniRefAPIModel } from '../../../uniref/adapters/uniRefConverter';
+} from '../../../uniprotkb/adapters/uniProtkbConverter';
+import { UniRefLiteAPIModel } from '../../../uniref/adapters/uniRefConverter';
 
-import { ViewMode } from '../../state/resultsInitialState';
+import { ViewMode } from '../../../uniprotkb/state/resultsInitialState';
 
-import apiUrls, { getAPIQueryUrl } from '../../../shared/config/apiUrls';
-import ColumnConfiguration from '../../config/ColumnConfiguration';
+import apiUrls, { getAPIQueryUrl } from '../../config/apiUrls';
+import UniProtKBColumnConfiguration from '../../../uniprotkb/config/UniProtKBColumnConfiguration';
+import UniRefColumnConfiguration, {
+  UniRefColumn,
+} from '../../../uniref/config/UniRefColumnConfiguration';
 
-import useDataApi from '../../../shared/hooks/useDataApi';
-import useNS from '../../../shared/hooks/useNS';
+import useDataApi from '../../hooks/useDataApi';
+import useNS from '../../hooks/useNS';
 
-import getNextUrlFromResponse from '../../utils/queryUtils';
+import getNextUrlFromResponse from '../../../uniprotkb/utils/queryUtils';
 import {
   getParamsFromURL,
   getLocationObjForParams,
   getSortableColumnToSortColumn,
-} from '../../utils/resultsUtils';
+} from '../../../uniprotkb/utils/resultsUtils';
 
-import { Namespace } from '../../../shared/types/namespaces';
-import { SortDirection, ReceivedFieldData } from '../../types/resultsTypes';
-import { SortableColumn, Column } from '../../types/columnTypes';
+import { Namespace } from '../../types/namespaces';
+import {
+  SortDirection,
+  ReceivedFieldData,
+} from '../../../uniprotkb/types/resultsTypes';
+import {
+  SortableColumn,
+  UniProtKBColumn,
+} from '../../../uniprotkb/types/columnTypes';
 
 import './styles/warning.scss';
 import './styles/results-view.scss';
+import { AllColumns } from './ResultsContainer';
 
 type ResultsTableProps = {
   selectedEntries: string[];
-  columns: Column[];
+  columns: AllColumns;
   viewMode: ViewMode;
   handleEntrySelection: (rowId: string) => void;
 };
 
-const ResultsView: React.FC<ResultsTableProps> = ({
+const convertRow = (
+  row: UniProtkbAPIModel | UniRefLiteAPIModel,
+  namespace: Namespace
+) => {
+  switch (namespace) {
+    case Namespace.uniprotkb:
+      return uniProtKbConverter(row as UniProtkbAPIModel) as UniProtkbUIModel;
+    case Namespace.uniref:
+      return row as UniRefLiteAPIModel;
+    default:
+      return null;
+  }
+};
+
+const getIdKey = (namespace: Namespace) => {
+  switch (namespace) {
+    case Namespace.uniprotkb:
+      return ({ primaryAccession }: { primaryAccession: string }) =>
+        primaryAccession;
+    case Namespace.uniref:
+      return ({ id }: { id: string }) => id;
+    default:
+      return null;
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ColumnConfigurations: Partial<Record<Namespace, Map<any, any>>> = {
```

Comment:
Maybe not needed for now, but maybe add a `TODO:` comment to remove the `any, any` types eventually.
We need a Column type covering UniProtKBColumns, UniRefColumns, etc for the first any, and a Column renderer type with `label: string` and a render definition for the second any.

Reply:
Added comment